doc: add dark mode and other styling improvements

elfenermarcell opened the pull request:
This adds dark mode to the HTML manuals and also centers the appendix sections. The dark background is based on search.nixos.org

The dark mode is currently not quite ready, as `pkgs/misc/documentation-highlighter/mono-blue.css` will need to be changed as well as code blocks have bad contrast. If you have any ideas on how to change the higlighting colours for dark mode, please share.

Also, the "Note" and "Warning" sections don't have very good contrast either, I'm looking for suggestions for those too. (hopefully without making them look too different from their original light mode design)

@NixOS/documentation-team 
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

elfenermarcell commented:
And a question: The font is set as `font-family: 'Roboto', sans-serif;`, but it makes no attempt to actually get Roboto, so if you don't happen to have it installed you'll just see your default sans-serif font. Is this intentional?

fricklerhandwerk commented:
The missing font is almost certainly not intentional. Could you please use variables for the colors, naming them by purpose (e.g. font, background, highlight, ...) ? We'd be missing out on one of the most important accomplishments of computing history otherwise.

elfenermarcell committed:
doc, documentation-highlighter: format css

elfenermarcell committed:
doc: center align appendices

elfenermarcell committed:
doc, documentation-highlighter: use variables & add dark mode

fricklerhandwerk reviewed (approved):
LGTM on cursory review. (I personally prefer a screenshot of what it should be like, but this looks innocuous enough of a change that I trust it will be alright.)

elfenermarcell committed:
doc, nixos/doc: include roboto font

Roboto was already set as the font for the manuals,
but without a way to get it, so people who didn't
happen to have it installed saw their default font

elfenermarcell commented:
I have taken some screenshots:
![image](https://github.com/user-attachments/assets/f907831f-860f-415c-918d-480da1fb6e84)
![image](https://github.com/user-attachments/assets/30e2b4b9-28d7-483c-86ef-66b10d3daf68)
![image](https://github.com/user-attachments/assets/3cc5955b-f5ab-4876-8ef5-8eb9183b559b)
![image](https://github.com/user-attachments/assets/cf806d4e-0fa0-4279-9c50-258a6c86c34f)

fricklerhandwerk commented:
Thanks a lot, great job! :)

drupol commented:
The Roboto font has been updated from version 2 to 3 in https://github.com/NixOS/nixpkgs/pull/389195 and the font's `web` directory doesn't exist any longer. Therefore, the following commands might be copying something that doesn't exist any longer:

```
cp ${pkgs.roboto.src}/web/Roboto\[ital\,wdth\,wght\].ttf "$dst/Roboto.ttf"
```
